packages/chatbot-server-mongodb-public/src/processors/findContentWithMongoDbMetadata.test.ts

```diff
@@ -0,0 +1,92 @@
+// Mocks
+jest.mock("mongodb-rag-core/mongoDbMetadata", () => {
+  const actual = jest.requireActual("mongodb-rag-core/mongoDbMetadata");
+  return {
+    ...actual,
+    classifyMongoDbProgrammingLanguageAndProduct: jest.fn(),
+  };
+});
+
+jest.mock("mongodb-rag-core", () => {
+  const actual = jest.requireActual("mongodb-rag-core");
+  return {
+    ...actual,
+    updateFrontMatter: jest.fn(),
+  };
+});
+
+jest.mock("mongodb-rag-core/braintrust", () => {
+  const actual = jest.requireActual("mongodb-rag-core/braintrust");
+  return {
+    ...actual,
+    wrapTraced: jest.fn(),
+  };
+});
+
+import { FindContentFunc, updateFrontMatter } from "mongodb-rag-core";
+import { wrapTraced } from "mongodb-rag-core/braintrust";
+import {
+  makeFindContentWithMongoDbMetadata,
+} from "./findContentWithMongoDbMetadata";
+import { classifyMongoDbProgrammingLanguageAndProduct } from "mongodb-rag-core/mongoDbMetadata";
+
+
+const mockedClassify =
+  classifyMongoDbProgrammingLanguageAndProduct as jest.Mock;
+const mockedUpdateFrontMatter = updateFrontMatter as jest.Mock;
+const mockedWrapTraced = wrapTraced as jest.Mock;
+
+function makeMockFindContent(result: string[]): FindContentFunc {
+  return jest.fn().mockResolvedValue(result);
+}
+
+afterEach(() => {
+  jest.resetAllMocks();
+});
+
+describe("makeFindContentWithMongoDbMetadata", () => {
+  test("enhances query with front matter and classification", async () => {
+    const inputQuery = "How do I use MongoDB with TypeScript?";
+    const expectedQuery = `---
+    product: driver
+    programmingLanguage: typescript
+    ---
+    How do I use MongoDB with TypeScript?`;
```

Comment:
this would have a weird indent. pls adjust all the multi-line strings accordingly

```suggestion
product: driver
programmingLanguage: typescript
---
How do I use MongoDB with TypeScript?`;
```